data/data/openstack/main.tf

```diff
@@ -70,22 +72,23 @@ module "masters" {
 module "topology" {
   source = "./topology"
 
-  cidr_block          = var.machine_v4_cidrs[0]
-  cluster_id          = var.cluster_id
-  cluster_domain      = var.cluster_domain
-  external_network    = var.openstack_external_network
-  external_network_id = var.openstack_external_network_id
-  masters_count       = var.master_count
-  api_floating_ip     = var.openstack_api_floating_ip
-  ingress_floating_ip = var.openstack_ingress_floating_ip
-  api_int_ip          = var.openstack_api_int_ip
-  ingress_ip          = var.openstack_ingress_ip
-  external_dns        = var.openstack_external_dns
-  trunk_support       = var.openstack_trunk_support
-  octavia_support     = var.openstack_octavia_support
-  machines_subnet_id  = var.openstack_machines_subnet_id
-  machines_network_id = var.openstack_machines_network_id
-  master_extra_sg_ids = var.openstack_master_extra_sg_ids
+  cidr_block            = var.machine_v4_cidrs[0]
+  cluster_id            = var.cluster_id
+  openstack_disable_sg  = var.openstack_disable_sg
+  cluster_domain        = var.cluster_domain
+  external_network      = var.openstack_external_network
+  external_network_id   = var.openstack_external_network_id
+  masters_count         = var.master_count
+  api_floating_ip       = var.openstack_api_floating_ip
+  ingress_floating_ip   = var.openstack_ingress_floating_ip
+  api_int_ip            = var.openstack_api_int_ip
+  ingress_ip            = var.openstack_ingress_ip
+  external_dns          = var.openstack_external_dns
+  trunk_support         = var.openstack_trunk_support
+  octavia_support       = var.openstack_octavia_support
+  machines_subnet_id    = var.openstack_machines_subnet_id
+  machines_network_id   = var.openstack_machines_network_id
+  master_extra_sg_ids   = var.openstack_disable_sg ? null : var.openstack_master_extra_sg_ids
```

Comment:
Do we need to enforce that null is used for the extra security group IDs? With `openstack_disable_sg` being set to true, `master_extra_sg_ids` will not be used in the module anyway.